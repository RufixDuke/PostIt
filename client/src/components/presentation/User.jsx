import React from 'react';
import PropTypes from 'prop-types';

/**
 * SearchedUser
 * @return {*} div
 */
const User = ({ user }) => {
  return (
    <div className="card search-user center green-text text-darken-4">
      <span>Username: {user.username}</span>
      <br />
      <span>Email: {user.email}</span>
      <br />
      <span>Phone Number: {user.phoneNumber}</span>
    </div>
  );
};

User.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string,
    email: PropTypes.string,
    phoneNumber: PropTypes.string
  }).isRequired
};

export default User;
