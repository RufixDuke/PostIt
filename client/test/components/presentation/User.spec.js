import React from "react";
import { mount } from "enzyme";
import User from '../../../src/components/presentation/User';

describe("User Component", () => {
  let props;
  let mountedUser;
  const user = {
    username: 'user1',
    email: 'user1@gmail.com',
    phoneNumber: '08055123456'
  };

  const userComponent = () => {
    if (!mountedUser) {
      mountedUser = mount(
        <User {...props} />
      );
    }
    return mountedUser;
  }

  beforeEach(() => {
    props = {
      user
    };
  });

  it("should always render div", () => {
    const divs = userComponent().find("div");
    expect(userComponent()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  describe("the rendered div", () => {
    it("should contain everything else that gets rendered", () => {
      const divs = userComponent().find("div");
      const wrappingDiv = divs.first();
      expect(wrappingDiv.children()).toEqual(userComponent().children());
    });
  });

  describe("rendered user", () => {
    it("should receive props", () => {
      expect(Object.keys(userComponent().props()).length).toBe(1);
    });
  });

  describe("when user prop is defined", () => {
    it("should display the username", () => {
      expect(userComponent().find('.username').text()).toBe("Username: user1");
    });
    it("should display the email", () => {
      expect(userComponent().find('.email').text()).toBe("Email: user1@gmail.com");
    });
    it("should display the phonenumber", () => {
      expect(userComponent().find('.phoneNumber').text()).toBe("Phone Number: 08055123456");
    });
  });
});
