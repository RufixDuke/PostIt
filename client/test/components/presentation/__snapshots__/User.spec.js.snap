// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component should always render div 1`] = `
<User
  user={
    Object {
      "email": "user1@gmail.com",
      "phoneNumber": "08055123456",
      "username": "user1",
    }
  }
>
  <div
    className="card search-user center green-text text-darken-4"
  >
    <span
      className="username"
    >
      Username: 
      user1
    </span>
    <br />
    <span
      className="email"
    >
      Email: 
      user1@gmail.com
    </span>
    <br />
    <span
      className="phoneNumber"
    >
      Phone Number: 
      08055123456
    </span>
  </div>
</User>
`;
